compiler/rustc_hir_typeck/src/fn_ctxt/adjust_fulfillment_errors.rs

```diff
@@ -477,12 +477,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // This is the "trait" (meaning, the predicate "proved" by this `impl`) which provides the `Self` type we care about.
         // For the purposes of this function, we hope that it is a `struct` type, and that our current `expr` is a literal of
         // that struct type.
-        let impl_trait_self_ref: Option<ty::TraitRef<'tcx>> =
-            self.tcx.impl_trait_ref(obligation.impl_def_id).map(|impl_def| impl_def.skip_binder());
-
-        let Some(impl_trait_self_ref) = impl_trait_self_ref else {
-            // It is possible that this is absent. In this case, we make no progress.
-            return Err(expr);
+        let impl_trait_self_ref = if self.tcx.is_trait_alias(obligation.impl_def_id) {
+            self.tcx.mk_trait_ref(
```

Comment:
Does this handle the alias-to-an-alias case?

Either way, could you add a double-alias to the new test case file to prevent regressions?

Reply:
I can't see why an alias-to-an-alias *wouldn't* work?  Nevertheless I have tested and can confirm it does indeed pass... happy to open a new PR to add that to the test case if it's valuable... but feels a little irrelevant to me?

Reply:
No worries, I don't think it's that big of a deal. It was just since I missed this interaction (between the error span refinement / trait aliases) the first time, I was concerned my code was even more brittle than I had expected.

Reply:
(I don't think this deserves a follow-up pr personally)

Reply:
Okay, let's leave it at that then!  Thanks all :)